src/server/pfs/drive/driver.go

```diff
@@ -1071,6 +1072,11 @@ func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint6
 	if fileInfo.FileType == pfs.FileType_FILE_TYPE_NONE {
 		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
 	}
+	// We return NotFound if it's a regular file but its blocks do not match the
+	// given filter shard
+	if fileInfo.FileType == pfs.FileType_FILE_TYPE_REGULAR && len(blockRefs) == 0 && (filterShard != nil && filterShard.BlockNumber > 1) {
+		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
+	}
```

Comment:
I'm only checking `filterShard != nil` so I can check `filterShard.BlockNumber` safely.

And yeah the point of this code is that if, after filtering, you get a file with not blockrefs, then you just return "file not found".  That way jobs won't see empty files.